### ISSUE TITLE
Amazon EKS Tutorial - Change necessary for database connectivity

### ISSUE THREAD

**Issue by aggietallboy** [open-amt-cloud-toolkit/open-amt-cloud-toolkit#208]
At this step:
https://open-amt-cloud-toolkit.github.io/docs/2.11/Tutorials/Scaling/Kubernetes/deployingk8s-eks/#create-databases-and-schema

The database is not "connectable" -- it gives a timeout error:
In order to correct this, I had to go to the RDS Database to identify the AZ, and then change the corresponding route table for that AZ (eksctl-awe-eks-amt-cluster/PrivateRouteTableUSEAST1D) to use the igw instead of the NAT gateway to enable communication.

Perhaps I can change it back after everything else is set up, but at this point, not quite sure.

**Comment by bwendlandt-intel** [comment]
Thanks for pointing this out and having a workaround. It looks like you are correct.  AWS seems to have shifted their default configuration a bit and our old path no longer works.

After some quick testing, you can swap the rule back after you've finished setting up the databases/schema. if desired. It does not affect the services.   I'll do some more testing and create a PR to correct our documentation.

**Comment by aggietallboy** [comment]
Either y'all or AWS seems to have made a change overnight that stopped making this step necessary (I'm betting AWS!)

**Comment by aggietallboy** [comment]
Identify all the private subnets

$global:privSubnets = (aws ec2 describe-route-tables --filters "Name=tag:Name,Values=`"eksctl-$clusterName-cluster/PrivateRouteTable*`"")|ConvertFrom-Json

Create the db subnet group with ONLY the private subnets
(aws rds create-db-subnet-group --db-subnet-group-name private-$eksVPC --db-subnet-group-description "$($clusterName) Database Subnet Group" --subnet-id $privSubnets.routetables.associations.subnetid --output json)|ConvertFrom-Json


Find the network igw and nat gw (saving them so we can put them back)

$global:dbAZ = $dbInstance.DBInstances.AvailabilityZone
$global:upperAZ = $dbAZ.Replace('-','').ToUpper()
$global:pubRTB = (aws ec2 describe-route-tables --filters "Name=tag:Name,Values=`"eksctl-$clusterName-cluster/PublicRouteTable`"")|ConvertFrom-Json
$global:privRTB = (aws ec2 describe-route-tables --filters "Name=tag:Name,Values=`"eksctl-$clusterName-cluster/PrivateRouteTable$($upperAZ)`"")|ConvertFrom-Json
$global:privNATGW = $privRTB.RouteTables.Routes.NatGatewayId
$global:pubIGW = ($pubrtb.RouteTables.Routes| Where-Object -Property DestinationCidrBlock -eq "0.0.0.0/0").GatewayId
$global:privRTBid = $privRTB.RouteTables.RouteTableId
$global:pubRTBid = $pubRTB.RouteTables.RouteTableId

Open up the subnet where the instance is homed:
(aws ec2 replace-route --route-table-id $privRTBid --destination-cidr-block '0.0.0.0/0' --gateway-id $pubigw)|ConvertFrom-Json


Close the subnet back up after we've created the tables:
(aws ec2 replace-route --route-table-id $privRTBid --destination-cidr-block '0.0.0.0/0' --nat-gateway-id $privNATGW)|ConvertFrom-Json




My database connectivity issue hasn't gone away yet, but this will force the database to "land" in the private subnets, and allow connectivity and then close it back up.

**Comment by aggietallboy** [comment]
after much head beating, and an open support case, I realized that the subnet security group, when we build it by hand, instead of automatically from the wizards do not include the cluster subnet

```
$global:secGrp = $dbInstance.DBInstances.VpcSecurityGroups.VpcSecurityGroupId
$global:clusterSG = ((aws ec2 describe-security-groups --filters "Name=tag:Name,Values=`"eks-cluster-sg-AWE-EKS*`"")|ConvertFrom-Json).SecurityGroups.Groupid
wog "Authorizing Groups"
wob (aws ec2 authorize-security-group-ingress --group-id $secGrp --ip-permissions "IpProtocol=tcp,FromPort=5432,ToPort=5432,IpRanges=$($ipRange)")|ConvertFrom-Json|ConvertTo-Json -Depth 32
wob (aws ec2 authorize-security-group-ingress --group-id $secGrp --source-group $clusterSG --protocol tcp --port 5432)|ConvertFrom-Json|ConvertTo-Json -Depth 32

```

Seems to have solved it.

Doing a (hopefully) final run through before posting the latest revision of the powershell script that addresses this.

**Comment by bwendlandt-intel** [comment]
Hello!  

I'm closing this issue.  I believe this was resolved offline a bit back.

Outside of the script vs RDS Create (in AWS UI) discrepancy mentioned above, the connection string had to change from `sslmode=no-verify` to `sslmode=require` exclusively for Postgres 15 DB instances.  Currently, AWS still defaults to Postgres 14, but we have modified the Open AMT documentation to include these details too [in this PR](https://github.com/open-amt-cloud-toolkit/docs/pull/307).  [Live page found here.](https://open-amt-cloud-toolkit.github.io/docs/2.12/Tutorials/Scaling/Kubernetes/deployingk8s-eks/#4-database-connection-strings)  Thanks!
